h5pyd/config.py

```diff
@@ -39,51 +42,164 @@ def __init__(self, config_file=None, **kwargs):
                         continue
                     fields = s.split('=')
                     if len(fields) < 2:
-                        print("config file: {} line: {} is not valid".format(self._config_file, line_number))
+                        print(f"config file: {self._config_file} line: {line_number} is not valid")
                         continue
                     k = fields[0].strip()
                     v = fields[1].strip()
-                    self._cfg[k] = v
+                    if k == "complex_names":
+                        self.complex_names = v
+                    elif k == "bool_names":
+                        self.bool_names = v
+                    elif k == "track_order":
+                        self.track_order = v
+                    else:
+                        Config._cfg[k] = v
+
+        # add standard keys if not already picked up
+        for k in ("hs_endpoint", "hs_username", "hs_password", "hs_api_key"):
+            if k not in Config._cfg:
+                Config._cfg[k] = ""
+
         # override any config values with environment variable if found
-        for k in self._cfg.keys():
+        for k in Config._cfg.keys():
             if k.upper() in os.environ:
-                self._cfg[k] = os.environ[k.upper()]
+                Config._cfg[k] = os.environ[k.upper()]
 
-        # finally update any values that are passed in to the constructor
+        # update any values that are passed in to the constructor
         for k in kwargs.keys():
-            self._cfg[k] = kwargs[k]
+            Config._cfg[k] = kwargs[k]
+
+        # finally, set defaults for any expected keys that are not already set
+        for k in ("hs_endpoint", "hs_username", "hs_endpoint"):
+            if k not in Config._cfg:
+                Config._cfg[k] = None
+        if "bool_names" not in Config._cfg:
```

Comment:
It looks like `bool_names` and `complex_names` are intended to let the application running HSDS customize what boolean and complex values are called.

Since this doesn't do anything right now and isn't related to this set of changes, I think it would be a good idea to bring this out until it's fully implemented

Reply:
That's valid, but I think I'll leave it in for compatibility.  complex and bool types will be coming soon.